src/pretty/mod.rs

```diff
@@ -163,7 +164,25 @@ impl<'a> PrettyPrinter<'a> {
     }
 
     fn format_disabled(&'a self, node: &'a SyntaxNode) -> ArenaDoc<'a> {
-        self.arena.text(node.clone().into_text().to_string())
+        if let Some(text) = self
+            .source
+            .find(node.span())
+            .and_then(|linked_node| self.source.get(linked_node.range()))
+        {
+            self.arena.text(text)
+        } else {
+            self.arena.text(node.clone().into_text().to_string())
+        }
+    }
+
+    /// For leaf only.
+    fn convert_verbatim(&'a self, node: impl AstNode<'a>) -> ArenaDoc<'a> {
+        self.convert_verbatim_untyped(node.to_untyped())
+    }
+
+    /// For leaf only.
+    fn convert_verbatim_untyped(&'a self, node: &'a SyntaxNode) -> ArenaDoc<'a> {
+        self.arena.text(node.text().as_str())
```

Comment:
maybe add a debug assert here to prevent incorrect usage

Reply:
actually, the previously frequently used `trivia` is also based on this assertion.
What I didn't have time to point out in the description is that, compared with `trivia`, this `verbatim` has fewer checks for inline items, and may be more performant.

Reply:
unfortunately, we could not check whether the node is leaf.